include/sparrow/array_data.hpp

```diff
@@ -197,6 +197,58 @@ namespace sparrow
     template <class L>
     void swap(reference_proxy<L>& lhs, reference_proxy<L>& rhs);
 
+    /*
+     * Layout iterator class
+     */
+    template <class L, bool is_const>
+    class layout_iterator : public iterator_base
+    <
+        layout_iterator<L, is_const>,
+        mpl::constify_t<typename L::value_type, is_const>,
+        std::contiguous_iterator_tag,
```

Comment:
These asserts would fail because the `contiguous_iterator` concept requires that the reference of the iterator is a true reference and not a proxy. However, we know by the spec that the underlying data *is* contiguous, and we could detect this by simply reading the tag of the iterator.

Besides, the iterator concepts do not check only the tag to decide whether an iterator type satistfies the concept requirements.

Reply:
> However, we know by the spec that the underlying data is contiguous

It is the case of all the layouts (_e.g._ [Sparse Union](https://arrow.apache.org/docs/format/Columnar.html#sparse-union))?

Reply:
> However, we know by the spec that the underlying data is contiguous

It is the case of all the layouts (_e.g._ [Sparse Union](https://arrow.apache.org/docs/format/Columnar.html#sparse-union))?

Reply:
My understanding is that the iterator kind is actually dependent on the kind of layout? So it cannot be decided here?

Reply:
How about?
```suggestion
        std::random_access_iterator_tag,
```

Reply:
Indeed, we should read this from the layout.